tests/meta/workflow/test_analysis.py

```diff
@@ -566,3 +578,57 @@ def test_link_fastq_files_for_sample(
 
         # THEN broadcast linking of files
         assert "Linking: " in caplog.text
+
+
+def test_are_all_samples_control(analysis_store: Store, case_id: str) -> None:
+    """Tests that are_all_samples_control returns True if all samples in a case are controls."""
+
+    # GIVEN a case with all samples being positive controls
+    case: Case = analysis_store.get_case_by_internal_id(case_id)
+    for sample in case.samples:
+        sample.control = ControlOptions.POSITIVE
+
+    # WHEN checking if all samples are controls
+    # THEN the result should be True
+    assert are_all_samples_control(case) == True
+
+
+@pytest.mark.parametrize(
+    "sample_controls, expected_qos",
+    [
+        # Test case 1: All samples are positive controls
+        pytest.param(
+            [ControlOptions.POSITIVE, ControlOptions.POSITIVE, ControlOptions.POSITIVE],
+            SlurmQos.EXPRESS,
+            id="all_controls",
+        ),
+        # Test case 2: Mixed controls (positive, negative, empty)
+        pytest.param(
+            [ControlOptions.POSITIVE, ControlOptions.NEGATIVE, ControlOptions.EMPTY],
+            SlurmQos.NORMAL,
+            id="not_all_controls",
+        ),
+    ],
```

Comment:
There is nothing wrong with the way you did it, just that all other test follow this convention instead
```suggestion
    [
        (
            [ControlOptions.POSITIVE, ControlOptions.POSITIVE, ControlOptions.POSITIVE],
            SlurmQos.EXPRESS,
        ),
        (
            [ControlOptions.POSITIVE, ControlOptions.NEGATIVE, ControlOptions.EMPTY],
            SlurmQos.NORMAL,
        ),
    ],
    ids = ["all controls", "not all controls" ]
```

Reply:
Note: Don't commit this suggestion directly in the PR, do it locally so that Balck lints the file properly